_infra/helm/locust/locustfiles/locustfile.py

```diff
@@ -529,24 +529,34 @@ def perform_requests(self):
             grouping = request.get("grouping")
             expected_response_text = request.get("expected_response_text")
             expected_response_status = request.get("response_status", 200)
+            request_url = request["url"] if "url" in request else None
+            harvest_dict = {}
 
-            if self.response and "harvest_url" in request:
+            if self.response and "harvest" in request:
                 soup = BeautifulSoup(self.response.text, "html.parser")
-
-                for link in soup.find_all(id=request["harvest_url"]["id"]):
-                    if request["harvest_url"]["link_text"] in link.get_text():
-                        request_url = link.get("href")
-                        break
-                    logger.error(f"Unable to harvest url {request['harvest_url']}")
-                    self.interrupt()
-            else:
-                request_url = request["url"]
+                harvest_details = request["harvest"]
+
+                if harvest_details["type"] == "url":
+                    for link in soup.find_all(id=request["harvest"]["id"]):
+                        if request["harvest"]["link_text"] in link.get_text():
+                            request_url = link.get("href")
+                            break
+                        logger.error(f"Unable to harvest url {request['harvest']}")
+                        self.interrupt()
+
+                if harvest_details["type"] == "id":
+                    for id_name in harvest_details["ids"]:
+                        input_name = soup.find("input", attrs={"name": id_name})
+                        input_value = input_name.attrs.get("value")
+                        harvest_dict[id_name] = input_value
 
             if request["method"] == "GET":
-                self.response =self.get(request_url, grouping, expected_response_text, expected_response_status)
+                self.response = self.get(request_url, grouping, expected_response_text, expected_response_status)
             elif request["method"] == "POST":
                 request_url = self.response.url if request_url == "self" else request_url
-                response_data = request['data']
+                response_data = request["data"]
+                response_data.update(harvest_dict)
```

Comment:
Doesn't matter hugely but harvest_dict could be empty

Reply:
Good point. I'll add a check in just on the off chance.